backend/src/main/kotlin/app/ehrenamtskarte/backend/application/database/repos/ApplicationRepository.kt

```diff
@@ -13,48 +16,63 @@ import com.fasterxml.jackson.module.kotlin.KotlinModule
 import org.jetbrains.exposed.sql.select
 import org.jetbrains.exposed.sql.transactions.transaction
 import java.io.File
 import java.nio.file.Paths
+import java.security.SecureRandom
+import java.util.Base64
 
 object ApplicationRepository {
     fun persistApplication(
         applicationJson: JsonField,
+        applicationVerifications: List<ApplicationVerificationView>,
         regionId: Int,
         applicationData: File,
-        files: List<Part>
-    ) {
-        val newApplication = transaction {
-            ApplicationEntity.new {
-                this.regionId = EntityID(regionId, Regions)
-                this.jsonValue = toString(applicationJson)
-            }
-        }
+        files: List<Part>,
+    ): Pair<ApplicationEntity, List<ApplicationVerificationEntity>> {
+        return transaction {
+            val newApplication =
+                ApplicationEntity.new {
+                    this.regionId = EntityID(regionId, Regions)
+                    this.jsonValue = toString(applicationJson)
+                }
 
-        val project =
-            transaction {
-                (Projects innerJoin Regions).slice(Projects.columns).select { Regions.id eq regionId }
-                    .single().let { ProjectEntity.wrapRow(it) }
+            val random = SecureRandom.getInstanceStrong()
+            val verificationEntities = applicationVerifications.map {
+                val byteArray = ByteArray(64)
+                random.nextBytes(byteArray)
+                val key = Base64.getUrlEncoder().encodeToString(byteArray)
+                ApplicationVerificationEntity.new {
+                    this.applicationId = newApplication.id
+                    this.accessKey = key
+                    this.contactName = it.contactName
+                    this.organizationName = it.organizationName
+                    this.contactEmailAddress = it.contactEmailAddress
+                }
             }
 
-        val projectDirectory = File(applicationData, project.project)
-        val applicationDirectory = File(projectDirectory, newApplication.id.toString())
+            val project = (Projects innerJoin Regions).slice(Projects.columns).select { Regions.id eq regionId }
+                .single().let { ProjectEntity.wrapRow(it) }
 
-        try {
-            files.forEachIndexed { index, part ->
-                FileUtil.streamToFile(
-                    part.inputStream,
-                    File(applicationDirectory, "$index").absolutePath
-                )
-                File(applicationDirectory, "$index.contentType").writeText(part.contentType)
-            }
-        } catch (e: Exception) {
-            applicationDirectory.deleteRecursively()
-            transaction {
-                newApplication.delete()
+            val projectDirectory = File(applicationData, project.project)
+            val applicationDirectory = File(projectDirectory, newApplication.id.toString())
```

Comment:
What are the files/directories used for?

Reply:
For the attachments (certificates,etc.) of the application.